apisix/plugins/basic-auth.lua

```diff
@@ -125,7 +125,7 @@ do
     end
 end
 
-function _M.access(conf, ctx)
+function _M.rewrite(conf, ctx)
```

Comment:
Need a test case to check if the `ctx.consumer` set by this plugin can be used by other plugin runs in access phase.

Reply:
Happy to add a test case. I don't have much experience writing them yet, so if you could point to a similar example, that would be helpful.

Reply:
You can add a stub:
```diff
diff --git apisix/init.lua apisix/init.lua
index 7333dd1..411cdd4 100644
--- apisix/init.lua
+++ apisix/init.lua
@@ -514,6 +514,10 @@ function _M.http_access_phase()
                 api_ctx.consumer,
                 api_ctx
             )
+
+            core.log.info("find consumer ", api_ctx.consumer.username,
+                          ", config changed: ", changed)
+
             if changed then
                 core.table.clear(api_ctx.plugins)
                 api_ctx.plugins = plugin.filter(route, api_ctx.plugins)
```

And then check it in:
https://github.com/apache/apisix/blob/e66dd9a314e0b545e412f806b8b8d98f3b1a8f2d/t/plugin/basic-auth.t#L192

Add something like:
```
--- error_log
find consumer foo
```

Reply:
Ok, I think I've seen an example in the test cases for `key-auth`. I'll try to add test cases acceding for `basic-auth`.